wrench/src/scene.rs

```diff
@@ -5,6 +5,76 @@
 use std::collections::HashMap;
 use webrender::api::{BuiltDisplayList, ColorF, Epoch};
 use webrender::api::{LayerSize, PipelineId};
+use webrender::api::{PropertyBinding, PropertyBindingId, LayoutTransform, DynamicProperties};
+
+/// Stores a map of the animated property bindings for the current display list. These
+/// can be used to animate the transform and/or opacity of a display list without
+/// re-submitting the display list itself.
+pub struct SceneProperties {
+    transform_properties: HashMap<PropertyBindingId, LayoutTransform>,
+    float_properties: HashMap<PropertyBindingId, f32>,
+}
+
+impl SceneProperties {
+    pub fn new() -> SceneProperties {
+        SceneProperties {
+            transform_properties: HashMap::default(),
+            float_properties: HashMap::default(),
+        }
+    }
+
+    /// Set the current property list for this display list.
+    pub fn set_properties(&mut self, properties: &DynamicProperties) {
+        self.transform_properties.clear();
+        self.float_properties.clear();
+
+        for property in &properties.transforms {
+            self.transform_properties
+                .insert(property.key.id, property.value);
+        }
+
+        for property in &properties.floats {
+            self.float_properties
+                .insert(property.key.id, property.value);
+        }
+    }
+
+    /// Get the current value for a transform property.
+    pub fn resolve_layout_transform(
```

Comment:
so why can't we share this code with the `scene.rs`? It's not complex, but still a few pages

Reply:
It's really just setting and getting from hash tables, and there are differences in the the signatures of the resolve methods due to the kind of references passed in. Also one piece of code uses warn! vs println!. Sharing it really just seems like a hassle.